### PR TITLE
fix: remove if-wrappers from operator error flags

### PR DESCRIPTION
**Pull request by toddbaert**
The `if` wrappers were added in [#356](https://github.com/open-feature/flagd-testbed/pull/356/commits/604df926b3b1c6caf5133a25e27c3bda87b2f90a) (my bad), but json-logic's `if` coerces its condition to a boolean; `null` and `false` are both falsy, so the `if` always selects the `"false"` variant for both cases. This makes it impossible to verify that operators return `null` (not `false`) on error, which is the entire point of the `@operator-errors` scenarios.

By using the operator directly as the top-level targeting expression, a `null` return propagates to the evaluator, which falls back to `defaultVariant` with reason `DEFAULT`; a `false` return would resolve to the `"false"` variant with reason `TARGETING_MATCH`. The distinction is preserved.

---

I tested this locally and with this and the expected fixes, I was able to get the test suite fully updated in in-process mode! No more inconsistencies!

### CODE REVIEW

File: evaluator/flags/testkit-flags.json [hunk 1]
```diff
@@ -462,38 +462,29 @@
...
-          {"sem_ver": [{"var": "version"}, "=", "1.0.0"]},
-          "true", "false"
-        ]
+        "sem_ver": [{"var": "version"}, "=", "1.0.0"]
       }
     },
     "semver-invalid-operator-flag": {
       "state": "ENABLED",
-      "variants": { "true": "true", "false": "false", "fallback": "fallback" },
+      "variants": { "match": "match", "fallback": "fallback" },
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply evaluator/flags/testkit-flags.json:473]: These are edge cases. We WANT to test for `null` returns.

File: evaluator/flags/testkit-flags.json [hunk 2]
```diff
@@ -462,38 +462,29 @@
     },
     "semver-invalid-version-flag": {
       "state": "ENABLED",
-      "variants": { "true": "true", "false": "false", "fallback": "fallback" },
+      "variants": { "match": "match", "fallback": "fallback" },
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply evaluator/flags/testkit-flags.json:465]: https://github.com/open-feature/flagd-testbed/pull/368#discussion_r3113481869

File: evaluator/flags/testkit-flags.json [hunk 3]
```diff
@@ -603,60 +594,45 @@
     },
     "starts-with-non-string-flag": {
       "state": "ENABLED",
-      "variants": { "true": "true", "false": "false", "fallback": "fallback" },
+      "variants": { "match": "match", "fallback": "fallback" },
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply evaluator/flags/testkit-flags.json:597]: https://github.com/open-feature/flagd-testbed/pull/368#discussion_r3113481869

File: evaluator/flags/testkit-flags.json [hunk 4]
```diff
@@ -603,60 +594,45 @@
...
-          {"ends_with": [{"var": "num"}, "xyz"]},
-          "true", "false"
-        ]
+        "ends_with": [{"var": "num"}, "xyz"]
       }
     },
     "starts-with-wrong-args-flag": {
       "state": "ENABLED",
-      "variants": { "true": "true", "false": "false", "fallback": "fallback" },
+      "variants": { "match": "match", "fallback": "fallback" },
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply evaluator/flags/testkit-flags.json:613]: https://github.com/open-feature/flagd-testbed/pull/368#discussion_r3113481869

File: flags/edge-case-flags.json [hunk 5]
```diff
@@ -56,128 +56,98 @@
...
+        "starts_with": [{"var": "num"}, "abc"]
       }
     },
     "ends-with-non-string-flag": {
       "state": "ENABLED",
       "variants": {
-        "true": "true",
-        "false": "false",
+        "match": "match",
         "fallback": "fallback"
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply flags/edge-case-flags.json:109]: https://github.com/open-feature/flagd-testbed/pull/368#discussion_r3113481869

File: flags/edge-case-flags.json [hunk 6]
```diff
@@ -56,128 +56,98 @@
...
+        "starts_with": ["abc"]
       }
     },
     "ends-with-wrong-args-flag": {
       "state": "ENABLED",
       "variants": {
-        "true": "true",
-        "false": "false",
+        "match": "match",
         "fallback": "fallback"
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply flags/edge-case-flags.json:131]: https://github.com/open-feature/flagd-testbed/pull/368#discussion_r3113481869
